### DISCUSSION TITLE
API for access to list object in backend

### DISCUSSION THREAD

**Discussion by MurzNN** [keystonejs/keystone#6124] [Event Q&A Q3 2021]
After Keystone's core was reworked by #5665, we miss the ability to access Lists metadata at backend side. 

For example, earlier we could get list of fields from needed List via something like this:
```js
const myListFields = context.lists[listName].fieldsByPath;
const myListAuthorFieldType = context.lists[listName].fieldsByPath['author'].adapter.fieldName; // eg 'Relationship'
```
Now this feature is disappeared, and I can't find any other way to get that information at the backend function.

**Comment by MurzNN** [comment]
Here https://keystonejs.slack.com/archives/C01STDMEW3S/p1625703474012900?thread_ts=1622762153.345400&cid=C01STDMEW3S is suggested to use this construction:
```
context.graphql.schema.getType(listKey).getFields()?.[fieldKey].extensions?.keystoneSecretField.compare(secret, hash)
```
But it not return all needed data about list field (type, database type, etc).

So I fix this via my PR here https://github.com/keystonejs/keystone/pull/6059

**Reply by gautamsi** [reply]
I need this too, have to make sure the output is formatted based on config
